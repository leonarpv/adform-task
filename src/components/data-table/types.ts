import { ReactNode } from 'react'

export type ColumnDefinition<TSource extends {}> = {
    accessor: (source: TSource) => ReactNode
    title: string
}
