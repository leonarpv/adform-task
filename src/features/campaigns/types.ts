export type Campaign = {
    id: number
    name: string
    startDate: string
    endDate: string
    Budget: number
    userId: number
}

export type CampaignUser = {
    id: number
    name: string
}

export type CampaignWithUser = {
    user: CampaignUser | null
} & Campaign

export type CampaignFilters = {
    name?: string
    startDate?: string
    endDate?: string
}
