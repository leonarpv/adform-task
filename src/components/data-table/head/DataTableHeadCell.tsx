import React from 'react'
import { ColumnDefinition } from '../types'

type Props<TSource> = {
    column: ColumnDefinition<TSource>
}

const DataTableHeadCell = <TSource extends {}>({ column }: Props<TSource>) => {
    const { title } = column

    return <th className="data-table__cell data-table__cell--head">{title}</th>
}

export default DataTableHeadCell
