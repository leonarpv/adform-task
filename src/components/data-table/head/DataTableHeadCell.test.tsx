import { shallow } from 'enzyme'
import React from 'react'
import DataTableHeadCell from './DataTableHeadCell'
import { Mock } from 'ts-mockery'
import {ColumnDefinition} from "../types";

describe('DataTableHeadCell', () => {
    it('should render properly', () => {
        const column = Mock.of<ColumnDefinition<string>>()

        const wrapper = shallow(<DataTableHeadCell column={column} />)

        expect(wrapper).toMatchSnapshot()
    })

    it('should render table head cell', () => {
        const column = Mock.of<ColumnDefinition<string>>({
            title: 'title'
        })

        const wrapper = shallow(<DataTableHeadCell column={column} />)

        expect(wrapper.find('.data-table__cell.data-table__cell--head').exists()).toBe(true)
        expect(wrapper.find('.data-table__cell.data-table__cell--head').text()).toBe(column.title)
    })
})
