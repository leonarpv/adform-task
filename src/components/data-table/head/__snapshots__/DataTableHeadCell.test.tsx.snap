// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTableHeadCell should render properly 1`] = `ShallowWrapper {}`;
